Devise related branch

Terunaga opened the pull request:
# WHAT

Set up some initial actions in order to implement the sign-up features.
Specifically, ran the following commands:
- rails g devise:install
- rails g devise user
  *rails g devise:views

Also, added the following columns to the users table:
- username
- member
- profile
- works
# WHY

These setups are crucial in implementing the sign-up features.

Terunaga committed:
Installed "devise" gem.

Terunaga committed:
Ran the following commands:
rails g devise:install
rails g devise user
rails g devise:views
rake db:migrate

Terunaga committed:
Added username, member, profile, and works columns to the users table.

sinagaki58 commented on app/views/devise/confirmations/new.html.erb:
```diff
@@ -0,0 +1,16 @@
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
erbは全てhamlにしましょう
erb2hamlというgemがあります

sinagaki58 replied:
空行も

sinagaki58 commented on app/views/devise/unlocks/new.html.erb:
```diff
@@ -0,0 +1,16 @@
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
deviseのview全て空行いれましょう

sinagaki58 commented on test/models/user_test.rb:
```diff
@@ -0,0 +1,7 @@
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
test消しましょう

sinagaki58 commented:
作業中のブランチだと思うので、その場合は
`[WIP]Devise related branch`
みたいな感じで[WIP]とつけましょう

Terunaga committed:
Deleted test related files.

Terunaga committed:
Ran rake html:erb2haml command, and created html file for each html.erb file.

Added a blank line to each view file.

Terunaga commented:
# WHAT
- Converted all the view files to html.haml extensions with rake haml:erb2haml command.
- Added a blank line to each view file created.

# WHY
- The view files will be created by html.haml extension.
- It gives a trouble without a blank line at the end of a file.

sinagaki58 commented:
lgtm

Terunaga committed:
Modified the link_to method, which caused a syntax error.

Terunaga committed:
Deleted devise files with html.erb extension.

Terunaga committed:
Modified the path to logout when "logout" button on the header is clicked.
Also, redirected to sign in page once logged out, with after_sign_out_path_for method.

Terunaga committed:
Added Username, Member, Profile, and Works labels on the registration page.

Terunaga committed:
Added a validation for username (one cannot register without a username).

Terunaga committed:
Added username, member, profile, and works as a parameter when registering.

Implemented with configure_permitted_parameters method, along with devise_parameter_sanitizer.

Terunaga committed:
Modified the sign_in page.

Terunaga committed:
Modified the sign up page.

Terunaga committed:
Installed carrierwave gem.

Terunaga committed:
Ran the command "rails g uploader Image".

Terunaga committed:
Created a new column for avatar with the following command, and ran rake db:migrate.
"rails g migration AddAvatarToUser avatar:string"

Terunaga committed:
Added mount_uploader.

Terunaga committed:
Added avatar to devise_parameter_sanitizer.

Terunaga committed:
Modified the header such that the buttons shown will be changed depending on whether a user is logged in or not.

Terunaga committed:
Installed the gem "pry-rails" .

Terunaga committed:
Added the link to my page and profile edit page respectively.

Terunaga committed:
Modified the profile edit view page.

Terunaga committed:
Deleted unnecessary actions (index, new, create) for user controller.

Terunaga committed:
Created show action for user controller.

Terunaga committed:
Deleted index action for user controller, and replaced its contents in show action for user controller.

Terunaga committed:
Added contents to update action such that modifications to profile will be updated.

Terunaga committed:
Acquired a user record for show action.

Also, added avatar to update_params.

Terunaga committed:
Test thumbnails.

Terunaga committed:
Modified the user profile page so that the user's avatar, username, profile and works will be shown.

Terunaga commented:
# WHAT

##【WIP】Devise related branch
- added gem "carrierwave".
- implemented user sign-up/sign-in/sign-out functions.
- added contents to update action of user controller.
- modified the view files for "my page", "edit', "sign in", "sign up" pages.

# WHY
- carrierwave gem is necessary for adding an avatar.
- a user can experience different features depending on whether a user has logged in or not.
- a user can edit his/her profile.
- some links were not added/haml wasn't functioning properly, etc.

sinagaki58 commented on app/controllers/application_controller.rb:
```diff
@@ -2,4 +2,15 @@ class ApplicationController < ActionController::Base
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.for(:sign_up).push(:username, :avatar, :member, :profile, :works)
+  end
```
account_updateも必要ですね

sinagaki58 commented on app/controllers/user_controller.rb:
```diff
@@ -10,12 +10,20 @@ def edit
+    @user = User.find(params[:id])
   end
 
   def update
```
editアクションも必要かと

sinagaki58 commented on app/views/layouts/_header.html.haml:
```diff
@@ -4,25 +4,26 @@
+        %ul.nav.navbar-nav.navbar-right
+          %li
+            .btn.btn-primary.navbar-btn
+              = link_to 'Get Started', new_user_session_path
```
空行

sinagaki58 commented on app/views/user/show.html.haml:
```diff
@@ -2,8 +2,8 @@
       .media-left
-        %a{href: "#"}
-          %img.media-object{alt: "64x64", data: {holder: {rendered: true}}, data: {src: "holder.js/64x64"}, src: "data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9InllcyI/PjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB3aWR0aD0iNjQiIGhlaWdodD0iNjQiIHZpZXdCb3g9IjAgMCA2NCA2NCIgcHJlc2VydmVBc3BlY3RSYXRpbz0ibm9uZSI+PGRlZnMvPjxyZWN0IHdpZHRoPSI2NCIgaGVpZ2h ...
+        %a{href: user_path(current_user)}
```
aタグはlink_toにしましょう

sinagaki58 commented:
画像はpushせずにgitignoreに入れましょう

Terunaga committed:
Modified the a tag to link_to method.

Terunaga committed:
Deleted unnecessary actions in user controller.

Terunaga committed:
Added a blank line.

Terunaga committed:
All files under public/uploads folder will be ignored.

Terunaga commented:
# WHAT
- modified <a> tag to link_to method.
- deleted unnecessary actions in user controller.
- added a blank line.
- put public/uploads folder in .gitignore.

# WHY
- link_to method is better than <a> tag.
- some actions are not needed.
- it gives a trouble without a blank line.
- to prevent pushing avatars.

sinagaki58 commented:
pushした画像消したらlgtmです

Terunaga committed:
Used git revert command.

Terunaga commented:
@sinagaki58 
Git revertで画像を消せたみたいです。
確認お願いします！
